ui-app/client/src/components/Timer/Timer.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { PageStoreExtractor, getPathFromLocation } from '../../context/StoreContext';
+import { ComponentProps } from '../../types/common';
+import { processComponentStylePseudoClasses } from '../../util/styleProcessor';
+import { runEvent } from '../util/runEvent';
+import useDefinition from '../util/useDefinition';
+import TimerStyle from './TimerStyle';
+import { propertiesDefinition, stylePropertiesDefinition } from './timerProperties';
+import { styleDefaults } from './timerStyleProperties';
+import { IconHelper } from '../util/IconHelper';
+import { HelperComponent } from '../HelperComponents/HelperComponent';
+import { SubHelperComponent } from '../HelperComponents/SubHelperComponent';
+
+function Timer(props: ComponentProps) {
+	const { definition, pageDefinition, locationHistory, context } = props;
+	const pageExtractor = PageStoreExtractor.getForContext(context.pageName);
+	const {
+		properties: { timerType, duration, initialDelay, repeatCount, onTimerEventFunction } = {},
+		stylePropertiesWithPseudoStates,
+	} = useDefinition(
+		definition,
+		propertiesDefinition,
+		stylePropertiesDefinition,
+		locationHistory,
+		pageExtractor,
+	);
+
+	const [output, setOutput] = useState<string>('');
+
+	const runUserDefinedFunction = useCallback(() => {
+		if (!onTimerEventFunction) return;
+		const event = pageDefinition.eventFunctions?.[onTimerEventFunction];
+		if (!event) return;
+		const result = runEvent(
+			event,
+			onTimerEventFunction,
+			context.pageName,
+			locationHistory,
+			pageDefinition,
+		);
+		setOutput(result?.toString() || '');
+	}, [pageDefinition, context.pageName, locationHistory, onTimerEventFunction]);
+
+	useEffect(() => {
+		const initialDelayMs = initialDelay || 0;
+		const durationMs = duration || 1000;
+
+		const initialDelayId = setTimeout(() => {
+			if (timerType === 'Non-Reapeating') {
+				runUserDefinedFunction();
+			} else {
+				let executionCount = 0;
+				const intervalId = setInterval(() => {
+					runUserDefinedFunction();
+					executionCount++;
+
+					if (repeatCount !== -1 && executionCount >= repeatCount) {
+						clearInterval(intervalId);
+					}
+				}, durationMs);
+
+				return () => clearInterval(intervalId);
+			}
+		}, initialDelayMs);
+
+		return () => clearTimeout(initialDelayId);
+	}, [timerType, duration, initialDelay, repeatCount, runUserDefinedFunction]);
```

Comment:
This logic won't work in the case of 
1. non-repeating when both initial delay and duration are given.
2. repeating case when the component is invisible it will continue to run it won't clear the interval.